test/models/residence_test.rb

```diff
@@ -10,4 +10,26 @@ def test_valid
     assert residence.valid?, residence.errors.messages
   end
 
+  test 'should display homeless instead of residence if address is homeless or none' do 
+    addr = addresses :homeless
+    assert_equal addr.full, 'Homeless'
+
+    # tests 'currently homeless'
+    residence = Residence.new address_id: addr.id, current: true
+    assert_match /homeless/i, residence.to_s
```

Comment:
Sorry about the delay getting these back -- this past week turned into a doozy for me. 

Changed these to all assert_equals from matches.